test/AdjudicationIntegration.t.sol

```diff
@@ -0,0 +1,438 @@
+
+import {L2ForkArbitrator} from "../contracts/L2ForkArbitrator.sol";
+import {L1GlobalRouter} from "../contracts/L1GlobalRouter.sol";
+import {L2ChainInfo} from "../contracts/L2ChainInfo.sol";
+
+import {MockPolygonZkEVMBridge} from "./testcontract/MockPolygonZkEVMBridge.sol";
+
+contract AdjudicationIntegrationTest is Test {
+    Arbitrator public govArb;
+
+    IERC20 internal tokenMock = IERC20(0x1234567890123456789012345678901234567890);
+
+    ForkableRealityETH_ERC20 internal l1realityEth;
+    RealityETH_v3_0 internal l2realityEth;
+
+    bytes32 internal addArbitratorQID1;
+    bytes32 internal addArbitratorQID2;
+    bytes32 internal removeArbitratorQID1;
+    bytes32 internal removeArbitratorQID2;
+    bytes32 internal upgradePropQID1;
+    bytes32 internal upgradePropQID2;
+
+    AdjudicationFramework internal adjudicationFramework1;
+    AdjudicationFramework internal adjudicationFramework2;
+
+    L2ForkArbitrator internal l2forkArbitrator;
+    L2ChainInfo internal l2ChainInfo;
+
+    Arbitrator internal l2Arbitrator1;
+    Arbitrator internal l2Arbitrator2;
+
+    address internal initialArbitrator1 = address(0xbeeb01);
+    address internal initialArbitrator2 = address(0xbeeb02);
+
+    address internal removeArbitrator1 = address(0xbabe05);
+    address internal removeArbitrator2 = address(0xbabe06);
+
+    address internal newForkManager1 = address(0xbabe07);
+    address internal newForkManager2 = address(0xbabe08);
+
+    address payable internal user1 = payable(address(0xbabe09));
+    address payable internal user2 = payable(address(0xbabe10));
+
+    string internal constant QUESTION_DELIM = "\u241f";
+
+    /*
+    Flow: 
+        - Add/remove arbitrator are requested via the bridge by an AdjudicationFramework on L2.
+        - Upgrade contracts are requested directly on L1, since L2 may be censored or non-functional.
+
+    TODO: Consider whether we should gate the realityeth instance to approved AdjudicationFramework contracts (via bridge) and an upgrade manager contract.
+    */
+
+    uint32 constant internal REALITY_ETH_TIMEOUT = 86400;
+
+    // Dummy addresses for things we message on l1
+    // The following should be the same on all forks
+    MockPolygonZkEVMBridge internal l2Bridge;
+    address internal l1GlobalRouter = address(0xbabe12);
+
+    // The following will change when we fork so we fake multiple versions here
+    address internal l1ForkingManager = address(0xbabe13);
+    address internal l1Token = address(0xbabe14);
+
+    address internal l1ForkingManagerF1 = address(0x1abe13);
+    address internal l1TokenF1 = address(0x1abe14);
+
+    address internal l1ForkingManagerF2 = address(0x2abe13);
+    address internal l1TokenF2 = address(0x2abe14);
+
+    uint32 internal l1chainId = 1;
+
+    uint256 internal forkingFee = 5000; // Should ultimately come from l1 forkingmanager
+
+    function setUp() public {
+
+        l2Bridge = new MockPolygonZkEVMBridge();
```

Comment:
Why can't you use hte formal mock functionality from foundry?

Reply:
I mean why is a costuom contract rerquired?

Reply:
Sorry, don't know how foundry mock functionality works, I'll take a look

Reply:
OK, I looked at this and I think the mock contract is clearer because it copies exactly what the real contract does direct from the real contract. I can recreate the same result with the Foundry stuff but then I have to do work to work out how to translate it, and whoever is reading the code has to do work to work out whether the way I translated it is correct.

Reply:
Do you have the code available? can I see it in another draft PR?

Reply:
You mean the code to do it the foundry way?

Reply:
yes

Reply:
Not gonna spend time futzing with that.